### PR TITLE
Fix bad typing on upload functions

### PR DESCRIPTION
**Pull request by Donokami**
## What kind of change does this PR introduce?

It simply update the function prototype of `upload` to type correctly the file argument

## What is the current behavior?

The upload function implements the management of Bytes/IO-like buffers, but the types of the functions doesn't allow their use.

### CODE REVIEW

File: storage3/_async/file_api.py [hunk 1]
```diff
@@ -164,7 +164,8 @@ async def list(
         """
         extra_options = options or {}
         extra_headers = {"Content-Type": "application/json"}
-        body = {**DEFAULT_SEARCH_OPTIONS, **extra_options, "prefix": path or ""}
+        body = {**DEFAULT_SEARCH_OPTIONS, **
+                extra_options, "prefix": path or ""}
```
Comment (anand2312) [inline storage3/_async/file_api.py:168]: Why the odd line-splits? It looked better before I think 😅

**Comment by anand2312** [comment]
Tests are failing because of the odd style changes introduced in this PR that Black (the code formatter) doesn't like. Please run Black on the changed files.

**Comment by Donokami** [comment]
Done!

**Comment by anand2312** [comment]
Thanks!

**Pull request merged**